proxy/src/serverless/sql_over_http.rs

```diff
@@ -244,11 +244,44 @@ pub(crate) async fn handle(
     backend: Arc<PoolingBackend>,
     cancel: CancellationToken,
 ) -> Result<Response<BoxBody<Bytes, hyper::Error>>, ApiError> {
+    let backend_clone = Arc::clone(&backend);
     let result = handle_inner(cancel, config, &ctx, request, backend).await;
 
     let mut response = match result {
         Ok(r) => {
             ctx.set_success();
+
+            // Handling the error response from local proxy here
+            if let crate::auth::Backend::ControlPlane(..) = backend_clone.auth_backend {
```

Comment:
This could be moved to a quick method in `impl Backend` like `is_cplane() -> bool`. But, generally, I'm a little concerned the proxy doesn't have a better way to tell what it is.

Reply:
Changed to is_local_proxy(), cause we were used the same flag below

Reply:
nit: technically, it's the auth backend that's local here not the proxy.